addons/pos_stripe/static/src/js/payment_stripe.js

```diff
@@ -193,10 +194,31 @@ let PaymentStripe = PaymentInterface.extend({
         }
     },
 
+    getCardBrandFromPaymentMethodDetails(paymentMethodDetails) {
+        // Both `card_present` and `interac_present` are "nullable" so we need to check for their existence, see:
+        // https://docs.stripe.com/api/charges/object#charge_object-payment_method_details-card_present
+        // https://docs.stripe.com/api/charges/object#charge_object-payment_method_details-interac_present
+        // In Canada `card_present` might not be present, but `interac_present` will be 
+        let brandElement;
+        if (paymentMethodDetails.card_present) {
+            brandElement = paymentMethodDetails.card_present;
+        } 
+        else if (paymentMethodDetails.interac_present) {
+            brandElement = paymentMethodDetails.interac_present;
+        }
+
+        if (brandElement) {
+            return brandElement.brand;
+        }
```

Comment:
what about:
```suggestion
        if (paymentMethodDetails.card_present) {
            return paymentMethodDetails.card_present;
        } 
        if (paymentMethodDetails.interac_present) {
            return paymentMethodDetails.interac_present;
        }
        return ""
```
this way you don't have to mutate a variable